src/game/game.cpp

```diff
@@ -529,6 +529,9 @@ void Game::saveGameState()
 
 	SPDLOG_INFO("Saving server...");
 
+	DBTransaction transaction;
+	transaction.begin();
```

Comment:
This makes a lot of sense, then we encapsulate the whole server save within a single transaction, meaning that we either save everything or nothing. The side effect is that this transaction will be relatively long, which could impact database performance. However, since server save happens without players, there is no load, thus the performance degradation won't be a huge factor here.

Reply:
This was made with an intention to force commit at end of server save, as autocommits are disabled. So in case of any poorer performance, you can remove all transaction lines from game.cpp and instead put code below under Map::save(), so it looks like this:

_Map::save();
Database& db = Database::getInstance();
db.executeQuery("commit;");_